pkg/spdk/disk.go

```diff
@@ -10,163 +9,26 @@ import (
-	if err != nil {
-		resp, parseErr := parseErrorMessage(err.Error())
-		if parseErr != nil || !isNoSuchDevice(resp.Message) {
-			return nil, errors.Wrapf(err, "failed to get lvstore with UUID %v", req.DiskUuid)
-		}
-		log.WithError(err).Errorf("Cannot find lvstore with UUID %v", req.DiskUuid)
-		return nil, grpcstatus.Errorf(grpccodes.NotFound, "cannot find lvstore with UUID %v", req.DiskUuid)
-	}
-
-	lvstore := &lvstores[0]
-
-	if lvstore.Name != req.DiskName {
-		log.Warnf("Disk name %v does not match lvstore name %v", req.DiskName, lvstore.Name)
-		return nil, grpcstatus.Errorf(grpccodes.NotFound, "disk name %v does not match lvstore name %v", req.DiskName, lvstore.Name)
-	}
-
-	_, err = s.spdkClient.BdevAioDelete(lvstore.BaseBdev)
-	if err != nil {
-		log.WithError(err).Errorf("Failed to delete AIO bdev %v", lvstore.BaseBdev)
-		return nil, errors.Wrapf(err, "failed to delete AIO bdev %v", lvstore.BaseBdev)
-	}
-	return &empty.Empty{}, nil
-}
-
-func (s *Server) DiskGet(ctx context.Context, req *spdkrpc.DiskGetRequest) (*spdkrpc.Disk, error) {
-	log := logrus.WithFields(logrus.Fields{
-		"diskName": req.DiskName,
-		"diskPath": req.DiskPath,
-	})
-
-	log.Info("Getting disk info")
-
-	if req.DiskName == "" || req.DiskPath == "" {
-		return nil, grpcstatus.Error(grpccodes.InvalidArgument, "disk name and disk path are required")
-	}
-
-	s.RLock()
-	defer s.RUnlock()
-
-	// Check if the disk exists
-	bdevs, err := s.spdkClient.BdevAioGet(req.DiskName, 0)
-	if err != nil {
-		resp, parseErr := parseErrorMessage(err.Error())
-		if parseErr != nil || !isNoSuchDevice(resp.Message) {
-			log.WithError(err).Errorf("Failed to get AIO bdev with name %v", req.DiskName)
-			return nil, grpcstatus.Errorf(grpccodes.Internal, errors.Wrapf(err, "failed to get AIO bdev with name %v", req.DiskName).Error())
-		}
-	}
-	if len(bdevs) == 0 {
-		log.WithError(err).Errorf("Cannot find AIO bdev with name %v", req.DiskName)
-		return nil, grpcstatus.Errorf(grpccodes.NotFound, "cannot find AIO bdev with name %v", req.DiskName)
-	}
-
-	diskPath := getDiskPath(req.DiskPath)
-
-	var targetBdev *spdktypes.BdevInfo
-	for i, bdev := range bdevs {
-		if bdev.DriverSpecific != nil ||
-			bdev.DriverSpecific.Aio != nil ||
-			bdev.DriverSpecific.Aio.FileName == diskPath {
-			targetBdev = &bdevs[i]
-			break
-		}
-	}
-	if targetBdev == nil {
-		log.WithError(err).Errorf("Failed to get AIO bdev name for disk path %v", diskPath)
-		return nil, grpcstatus.Errorf(grpccodes.NotFound, errors.Wrapf(err, "failed to get AIO bdev name for disk path %v", diskPath).Error())
-	}
-
-	return s.lvstoreToDisk(req.DiskPath, req.DiskName, "")
-}
-
 func getDiskPath(path string) string {
 	return filepath.Join(hostPrefix, path)
 }
 
-func getDeviceNumber(filename string) (string, error) {
-	fileInfo, err := os.Stat(filename)
+func getDiskID(filename string) (string, error) {
+	executor := util.NewTimeoutExecutor(util.CmdTimeout)
```

Comment:
nit: `util.CmdTimeout` is 1 min, but getting disk ID should be quick enough, so probably just use a no-timeout executor for this kind of operations.